src/backend/cuda/dims_param.hpp

```diff
@@ -0,0 +1,18 @@
+/*******************************************************
+ * Copyright (c) 2020, ArrayFire
+ * All rights reserved.
+ *
+ * This file is distributed under 3-clause BSD license.
+ * The complete license agreement can be obtained at:
+ * http://arrayfire.com/licenses/BSD-3-Clause
+ ********************************************************/
+
+#pragma once
+
+namespace cuda {
+
+typedef struct {
+    int dim[4];
+} dims_t;
```

Comment:
Yes, that was correct until recently. I am using Param structure now with memcpy and copy kernels and the tests pass fine without any issues. However, this `dims_t` is used else where in `kernel::copy` function to pass some extra dimensions (apart from Param/CParam) - an array of 4 dim_t elements.

I didn't have a strong opinion of retaining vs removing it, since there could be some future cases where such `dims_t` POD struct can be used by other kernels too. So, I kept it.

Reply:
I feel like this encourages its use which we should avoid. I think we should remove it from its own header.